admin/billing/payment/stripe_webhook.go

```diff
@@ -0,0 +1,139 @@
+package payment
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/stripe/stripe-go/v79"
+	"github.com/stripe/stripe-go/v79/webhook"
+	"go.uber.org/zap"
+)
+
+// handleWebhook handles incoming webhook events from Stripe
+func (s *Stripe) handleWebhook(w http.ResponseWriter, r *http.Request) {
+	const MaxBodyBytes = int64(65536)
+	r.Body = http.MaxBytesReader(w, r.Body, MaxBodyBytes)
+	payload, err := io.ReadAll(r.Body)
+	if err != nil {
+		s.logger.Error("error reading request body", zap.Error(err))
+		http.Error(w, "error reading request body", http.StatusServiceUnavailable)
+		return
+	}
```

Comment:
Consider using `httputil.Handler` and `httputil.Errorf` for returning errors.

Also consider using `observability.Middleware` for logging HTTP requests and errors instead of duplicating the error messages as logs (as happens in several cases here)

Reply:
yeah added here because observability middleware was not added, will remove this